pkg/pillar/utils/generics.go

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2023 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package utils
+
+// EqualSlices returns true if the two slices are equal in size
+// and items they contain.
+// This function can be used if slice items are comparable
+// (operator "==" can be used).
+func EqualSlices[Type comparable](list1, list2 []Type) bool {
```

Comment:
Why don't we use [reflect.DeepEqual](https://pkg.go.dev/reflect#DeepEqual) instead?

Reply:
DeepEqual uses reflection and therefore is quite heavy. It is mostly used in unit tests or when the programmer is lazy :)

Reply:
Okay, for me it was controversial, that reflection is heavier since golang is statically typed language and I thought it would be evaluated in compile time. But found this [answer](https://stackoverflow.com/questions/69909502/performance-of-using-reflect-to-access-struct-field-as-a-string-variable-vs-ac) explaining why it is slower. Thanks! :)